testutil/ibc_testing/generic_setup.go

```diff
@@ -0,0 +1,62 @@
+package ibc_testing
+
+import (
+	"testing"
+
+	ibctesting "github.com/cosmos/ibc-go/v3/testing"
+	"github.com/cosmos/interchain-security/testutil/e2e"
+)
+
+// Contains generic setup code for running e2e tests against a provider, consumer,
+// and/or democracy consumer app.go implementation. You should not need to modify or replicate this file
+// to run e2e tests against your app.go implementations!
+
+var (
+	provChainID          = ibctesting.GetChainID(1)
+	democConsumerChainID = ibctesting.GetChainID(2)
+)
+
+// AddProvider adds a new provider chain to the coordinator and returns the test chain and app type
+func AddProvider[T e2e.ProviderApp](coordinator *ibctesting.Coordinator, t *testing.T, appIniter ibctesting.AppIniter) (
+	*ibctesting.TestChain, T) {
+
+	provider := ibctesting.NewTestChain(t, coordinator, appIniter, provChainID)
+	coordinator.Chains[provChainID] = provider
+
+	return provider, provider.App.(T)
+}
+
+// AddDemocracyConsumer adds a new democ consumer chain to the coordinator and returns the test chain and app type
+func AddDemocracyConsumer[T e2e.DemocConsumerApp](coordinator *ibctesting.Coordinator, t *testing.T,
+	appIniter ibctesting.AppIniter) (*ibctesting.TestChain, T) {
+
+	democConsumer := ibctesting.NewTestChain(t, coordinator, appIniter, democConsumerChainID)
```

Comment:
Why is democConsumer setup different from initialization done in `AddConsumers`?

Provider's valset is copied to each of the instantiated consumers so I'm a bit confused. Isn't democracy chain intended to be used as a CCV consumer?

Reply:
It's notable that the democracy consumer e2e tests are [separated from](https://github.com/cosmos/interchain-security/blob/4c96805c51192a4ce9f104119d30400e4d122fe2/tests/e2e/democracy.go) the normal e2e tests. The democracy consumer tests do not run with a provider, so they need to be setup slightly differently (no copying of provider val set).

The democracy consumer should be able to pass the normal set of e2e ccv tests tho, it's not currently able to do this. see https://github.com/cosmos/interchain-security/issues/397